server/aws-lsp-codewhisperer/src/language-server/telemetryService.ts

```diff
@@ -80,10 +89,43 @@ export class TelemetryService extends CodeWhispererServiceToken {
         this.sendTelemetryEvent(request)
     }
 
-    public emitUserTriggerDecision(session: CodeWhispererSession, timeSinceLastUserModification?: number) {
-        if (this.shouldNotSendTelemetry()) {
-            return
+    private getCWClientTelemetryInteractionType(interactionType: ChatInteractionType): ChatMessageInteractionType {
+        let chatMessageInteractionType: ChatMessageInteractionType
+        switch (interactionType) {
+            case ChatInteractionType.InsertAtCursor:
+                chatMessageInteractionType = 'INSERT_AT_CURSOR'
```

Comment:
super nit: you can save few lines if you return immediately or convert this function to a map between values